smslibrary/src/main/java/com/eis0/smslibrary/SMSManager.java

```diff
@@ -6,23 +6,17 @@
 import android.content.Intent;
 import android.content.IntentFilter;
 
-public class SMSManager extends CommunicationHandler<SMSMessage> {
+/**
+ * Manager class of the library. It is the highest level, it communicates with external activities.
+ *
+ * @author Marco Cognolato
+ * @author Matteo Carnelos
+ */
+public class SMSManager extends CommunicationManager<SMSMessage> {
+
     // Singleton Design Pattern
     private SMSManager() { }
     private static SMSManager instance = null;
-    private Context context;
-
-    /**
-     * Returns an instance of SMSManager if none exist, otherwise the one instance already created
-     * as per the Singleton Design Patter, gets also the context of the application for future use
-     * @param context Context of the application to use when needed
-     * @return Single instance of this class
-     */
-    public static SMSManager getInstance(Context context) {
-        if(instance == null) instance = new SMSManager();
-        instance.context = context;
-        return instance;
-    }
```

Comment:
Te le faccio in italiano la review e le domande; robba nostra questa.
Come mai prima gestivi il contesto dell'istanza, e ora non più? Non era necessario fin dall'inizio e te ne sei accorto ora, o hai trovato un modo per non doverlo richiedere?

Reply:
Ah, ho usato il tasto "start a review". Devo far pratica anche di questo

Reply:
La creazione di SMSManager con il context richiesto nel costruttore è stata fatta da @Cogno-IDU. Io mi sono accorto che nell'utilizzare la libreria era scomodo ogni volta tirare fuori il context che serviva solo per i metodi che ti ritornavano una ricevuta di invio/ricezione, che non usavamo mai tra l'altro. Quindi ho spostato la richiesta del context direttamente dentro quei metodi.

Reply:
Mhhh, ok. Questo mi fa venire in mente un paio di "errori" miei passati. O meglio, ho pensato qualche metodo in maniera scorretta. Apposto